### DISCUSSION TITLE
.toHaveTableContent()

### DISCUSSION THREAD

**Discussion by jimbolla-aurora** [testing-library/jest-dom#389] [Ideas]
I'd like to be able to do something like:

```js

        expect(screen.getByRole('table')).toHaveTableContent([
            ['#', /movie/i, /year/i, /rating/i],
            ['1.', /shawshank/i, '1994', '9.2',],
            ['2.', /godfather/i, '1972', '9.1'],
            ['3.', /godfather: part ii/i, '1974', '9.0'],
            ['4.', /dark knight/i, '2008', '9.0'],
            ['5.', /12 angry men/i, '1957', '8.9']
        ]);
```

and then even cooler:

```js
        expect(screen.getByRole('table')).toHaveTableContent(fromMDTable`
            |   # | ${/movie/i}              | ${/year/i} | ${/rating/i} |
            | --- | ------------------------ | ---------- | ------------ |
            |   1 | ${/shawshank/i}          |       1994 |          9.2 |
            |   2 | ${/godfather/i}          |       1972 |          9.1 |
            |   3 | ${/godfather: part ii/i} |       1974 |          9.0 |
            |   4 | ${/dark knight/i}        |       2008 |          9.0 |
            |   5 | ${/12 angry men/i}       |       1957 |          8.9 |
        `);
```

**Comment by gnapse** [comment]
Interesting idea. Unfortunately, I do not have any bandwidth to implement this. Also, it's a non-trivial piece of work if you consider that tables can have a pretty varied set of layouts (e.g. multiple header rows, footer rows, cells that span across multiple rows or across multiple columns, etc.)

Not saying it is not possible, but I wonder how to deal with those situations in a generic way. If you're dealing with specific and more predictable tables that you'd like to assert against in an app you're working on, I'd recommend that you write your own custom matcher that can be less generic and more targeted to your specific use case. But doing this in a way that supports any table that you may throw at it can be daunting.
